yt_dlp/extractor/laxarxames.py

```diff
@@ -0,0 +1,77 @@
+import json
+
+from .brightcove import BrightcoveNewIE
+from .common import InfoExtractor
+from ..utils import ExtractorError
+from ..utils.traversal import traverse_obj
+
+
+class LaXarxaMesIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?laxarxames\.cat/(?:[^/]+/)*?(player|movie-details)/(?P<id>\d+)'
+    _NETRC_MACHINE = 'laxarxames'
+    _TOKEN = None
+    _LOGIN_URL = 'https://www.laxarxames.cat/login'
+    _TESTS = [{
+        'url': 'https://www.laxarxames.cat/player/3459421',
+        'md5': '0966f46c34275934c19af78f3df6e2bc',
+        'info_dict': {
+            'id': '6339612436112',
+            'ext': 'mp4',
+            'title': 'Resum | UA Horta — UD Viladecans',
+            'timestamp': 1697905186,
+            'thumbnail': r're:https?://.*\.jpg',
+            'description': '',
+            'upload_date': '20231021',
+            'duration': 129.44,
+            'tags': ['ott', 'esports', '23-24', ' futbol', ' futbol-partits', 'elit', 'resum'],
+            'uploader_id': '5779379807001',
+        },
+        'skip': 'Requires login',
+    }]
+
+    def _perform_login(self, username, password):
+        if self._TOKEN:
+            return
+        login = self._download_json(
+            'https://api.laxarxames.cat/Authorization/SignIn', None, note='Logging in', headers={
+                'X-Tenantorigin': 'https://laxarxames.cat',
+                'Content-Type': 'application/json',
+            }, data=json.dumps({
+                'Username': username,
+                'Password': password,
+                'Device': {
+                    'PlatformCode': 'WEB',
+                    'Name': 'Mac OS ()',
+                },
+            }).encode('utf-8'),
+            expected_status=401,
+        )
+
+        if not traverse_obj(login, ('AuthorizationToken', 'Token')):
+            raise ExtractorError('Login failed', expected=True)
+        else:
+            self._TOKEN = login['AuthorizationToken']['Token']
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        if not self._TOKEN:
+            self.raise_login_required()
+        mediaplayinfo = self._download_json(
```

Comment:
if adopting the site/API var names, convert camelCase/PascalCase var names to snake_case
```suggestion
        video_id = self._match_id(url)
        if not self._TOKEN:
            self.raise_login_required()

        media_play_info = self._download_json(
```